### PR TITLE
use free_desktop_entry to search desktop file

### PR DESCRIPTION
**Pull request by wiiznokes**
This depend on https://github.com/pop-os/freedesktop-desktop-entry/pull/15.

This will have the benefit of
- one api to get all desktop files
- cache them in your struct without a wrapper
- one api to search desktop files
- localisation support for searching entries

Currently, this works really great. The searching function still need refinement, like decide what fields to include in the search, give some field more weight, ect. But any improvement will benefit all cosmic component now.

Also, idk it it's possible yet, but it will be nice to give the returned response a score so that good matches get displayed first. I think this should be managed in the frontend since there can be multiple plugin at once, but i'm not sure.

### CODE REVIEW

File: plugins/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,9 @@ publish = false
 [dependencies]
 async-pidfd = "0.1.4"
 fork = "0.1.23"
-freedesktop-desktop-entry = "0.5.2"
```
Comment (mmstick) [inline plugins/Cargo.toml:12]: This can be set to 0.5.3

**Comment by wiiznokes** [comment]
I think having a git dependencie for fde is fine here since it will likely require some change in the matching function while cosmic approaching its release

File: plugins/Cargo.toml (hunk 1 above)
Comment (mmstick) [reply plugins/Cargo.toml:12]: This is now 0.5.4

Comment (wiiznokes) [reply plugins/Cargo.toml:12]: Okay done

File: plugins/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ publish = false
...
```
Comment (mmstick) [inline plugins/Cargo.toml:12]: Must update to 0.6.0 now

**Review by mmstick — CHANGES_REQUESTED** [review]

File: plugins/Cargo.toml (hunk 2 above)
Comment (wiiznokes) [reply plugins/Cargo.toml:12]: Done

**Review by mmstick — APPROVED** [review]

**Review by jacobgkau — APPROVED** [review]
I'm not seeing any user-facing difference in search behavior while testing this.

<details>
<summary>Regression testing passed:</summary>

- [X] All windows on all workspaces appear on launch
- [X] Choosing an app on another workspace moves workspaces and focus to that app
- [X] Launching an application works
- [X] Typing text and then removing it will re-show those windows
- [X] Search works for applications and windows
- [X] Search works for GNOME settings panels
- [X] Search for "Extensions". There should be only one entry.
- [X] The overlay hint correctly highlights the selected window
- [X] Open windows are sorted above applications (e.g. "firefox")
- [X] t: executes a command in a terminal
- [X] : executes a command in sh
- [X] = calculates an equation
- [X] Search results are as expected:
    - `cal` returns Calendar and Calculator before Color
    - `pops` returns Popsicle first
    - `shop` returns the Pop!_Shop first
</details>

**Pull request merged**